temporal102/Exercise1.hs

```diff
@@ -0,0 +1,139 @@
+-- - No duplicate work is performed during failover
+
+-- | Input data for the hello/goodbye translation workflow
+data TranslationInput = TranslationInput
+  { name :: Text           -- Name to greet
+  , languageCode :: Text   -- Target language (fr, es, de, pt)
+  } deriving (Generic, FromJSON, ToJSON, Show)
+
+-- | Output data containing translated greetings
+data TranslationOutput = TranslationOutput
+  { helloMessage :: Text     -- Translated hello message
+  , goodbyeMessage :: Text   -- Translated goodbye message
+  } deriving (Generic, FromJSON, ToJSON, Show)
+
+-- | Translation dictionary mapping language codes to term translations
+-- Supports French (fr), Spanish (es), and German (de)
+translations :: Map.Map Text (Map.Map Text Text)
+translations = Map.fromList
+  [ ("fr", Map.fromList [("hello", "bonjour"), ("goodbye", "au revoir")])
+  , ("es", Map.fromList [("hello", "hola"), ("goodbye", "adiós")])
+  , ("de", Map.fromList [("hello", "hallo"), ("goodbye", "auf wiedersehen")])
+  , ("pt", Map.fromList [("hello", "olá"), ("goodbye", "tchau")])
+  ]
+
+-- Activities are the basic unit of work in Temporal
+-- side effects and can be retried automatically on failure
+translateActivity :: (Text, Text) -> Activity () Text
+translateActivity (term, lang) = do
+  case Map.lookup lang translations >>= Map.lookup (T.toLower term) of
+    Just translation -> return translation
+    Nothing -> return term
+
+-- Register the activity with Temporal's code generation
+Temporal.TH.registerActivity 'translateActivity
+
+-- | Main workflow that demonstrates durable execution
+-- 
+-- This workflow:
+-- 1. Translates "hello" to the target language
+-- 2. Sleeps for 10 seconds (prime time to kill workers!)
+-- 3. Translates "goodbye" to the target language
+-- 4. Returns both translated messages
+--
+-- The 10-second sleep provides a window to test
+-- durable execution by killing workers during the sleep period.
+sayHelloGoodbyeWorkflow :: TranslationInput -> Workflow.Workflow TranslationOutput
+sayHelloGoodbyeWorkflow input = provideCallStack $ do
+  -- First activity: translate "hello"
+  hello <- Workflow.executeActivity TranslateActivity activityOptions ("hello", input.languageCode)
+  let helloMsg = hello <> ", " <> input.name
+  
+  -- 10-second sleep: perfect time to test worker failure recovery
+  Workflow.sleep (seconds 10)
+  
+  -- Second activity: translate "goodbye" 
+  goodbye <- Workflow.executeActivity TranslateActivity activityOptions ("goodbye", input.languageCode)
+  let goodbyeMsg = goodbye <> ", " <> input.name
+  
+  return $ TranslationOutput helloMsg goodbyeMsg
+  where
+    activityOptions = Workflow.defaultStartActivityOptions (Workflow.StartToClose $ seconds 30)
+
+-- Register the workflow with Temporal's code generation
+Temporal.TH.registerWorkflow 'sayHelloGoodbyeWorkflow
+
+-- Bring generated constructors into scope
+Temporal.TH.bringRegisteredTemporalFunctionsIntoScope
```

Comment:
@matto-mercury indicated that this might not have been strictly necessary in one of the other examples; did you get an error without this line?